utils/crypto/bls/signer/rpcsigner/client.go

```diff
@@ -46,22 +66,58 @@ func (c *Client) PublicKey() *bls.PublicKey {
 	return c.pk
 }
 
+// Sign a message. The [Client] already handles transient connection errors. If this method fails, it will
+// render the client in an unusable state and the client should be discarded.
 func (c *Client) Sign(message []byte) (*bls.Signature, error) {
+	var err error
+	defer func() {
+		if err != nil {
+			c.Close()
```

Comment:
Another idea I'm wondering is that if we can do better than the current ux of forcing the caller to re-create the client when this errors and manage that within the `Client` type. What if we started a goroutine in `New` that would keep trying to re-connect if we closed the connection, and `Sign*` if we're not connected could just fast-fail and return some "disconnected error"?